### PR TITLE
Cooperative cancellation sample

### PR DESCRIPTION
**Pull request by boblangley**
Introduces a sample to demonstrate cooperative cancellation of an endpoint and long running handler.

### CODE REVIEW

File: samples/endpoint-cancellation/Core_8/Server/Program.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Logging;
+
+class Program
+{
+    static async Task Main()
+    {
+        Console.Title = "Samples.FullDuplex.Server";
```
Comment (boblangley) [inline samples/endpoint-cancellation/Core_8/Server/Program.cs:11]:
```suggestion
        Console.Title = "Samples.Cooperative.Cancellation";
```

File: samples/endpoint-cancellation/Core_8/Server/Server.csproj [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>net5.0;netcoreapp3.1;netcoreapp2.1;net48</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="8.0.0-alpha.*" />
```
Comment (boblangley) [inline samples/endpoint-cancellation/Core_8/Server/Server.csproj:8]:
```suggestion
    <PackageReference Include="NServiceBus" Version="8.0.0-alpha.1897" />
```

File: samples/endpoint-cancellation/Core_8/Server/Program.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        LogManager.Use<DefaultFactory>()
+            .Level(LogLevel.Info);
+        var endpointConfiguration = new EndpointConfiguration("Samples.FullDuplex.Server");
```
Comment (boblangley) [inline samples/endpoint-cancellation/Core_8/Server/Program.cs:14]:
```suggestion
        var endpointConfiguration = new EndpointConfiguration("Samples.Cooperative.Cancellation");
```

File: samples/endpoint-cancellation/sample.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+title: Endpoint cancellation
```
Comment (boblangley) [inline samples/endpoint-cancellation/sample.md:2]:
```suggestion
title: Cooperative cancellation
```

File: samples/endpoint-cancellation/sample.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+summary: Stopping an endpoint with a cancellation token
```
Comment (boblangley) [inline samples/endpoint-cancellation/sample.md:3]:
```suggestion
summary: Stopping both the endpoint and a long running handler with a cancellation token
```

File: samples/cooperative-cancellation/sample.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+---
+title: Cooperative cancellation
+summary: Stopping both the endpoint and a long running handler with a cancellation token
+reviewed: 2021-08-31
+component: Core
+---
+
+
+Run the solution. One console application starts up, `Server`.
```
Comment (WilliamBZA) [inline samples/cooperative-cancellation/sample.md:9]: Skimming through the content I originally read "Once the console application starts up" and then got confused by the sentence. Changing it makes it less likely for the same skimming mistake to happen to others.

```suggestion
Run the solution. A single console application starts up: `Server`.
```

**Review by SzymonPobiega — APPROVED** [review]

**Review by WilliamBZA — APPROVED** [review]

**Pull request merged**